src/spid-sdk.js

```diff
@@ -11,6 +11,28 @@ var
     cache = require('./spid-cache'),
     talk = require('./spid-talk');
 
+var sessionCallbacks = (function sessionRequests() {
```

Comment:
Looking at this, it might be interesting to have it in it's own module.
Currently it only blocks concurrent session requests, not concurrent hasProduct requests.